### PR TITLE
refactor: 重构 Stats 插件初始化入参，支持自定义容器和是否可见

### PR DESCRIPTION
**Pull request by yiiqii**
## Release Notes

- **New Features**
	- Enhanced component management with the addition of optional parameters in various hooks and classes for improved configurability.
	- New methods `hide`, `show`, and `dispose` added to the `Stats` class for better control over component visibility and lifecycle.
	- Introduced debug functionality across multiple hooks for more flexible debugging control.

- **Bug Fixes**
	- Streamlined debugging process in several hooks by allowing direct control through constructor parameters.

- **Documentation**
	- Updated comments and method signatures to reflect new parameters and functionality across multiple classes.

### CODE REVIEW

File: plugin-packages/stats/src/monitor.ts [hunk 1]
```diff
@@ -109,7 +116,7 @@ export class Monitor implements Disposable {
   update (dt: number): void {
     const data = this.core.update(dt);
 
-    if (!data || data.drawCall === 0 || data.triangles === 0) {
+    if (!data || data.triangles === 0) {
```
Comment (yiiqii) [inline plugin-packages/stats/src/monitor.ts:119]: https://mdn.alipayobjects.com/mars/afts/file/A*P4lHTKYL_DgAAAAAAAAAAAAADlB4AQ
这种简单的合成 drawCall 一直是 0，这里的判断会导致后面的逻辑不会继续计算
@Sruimeng 看看去掉是否合理

File: packages/effects-core/src/vfx-item.ts [hunk 2]
```diff
@@ -315,10 +315,15 @@ export class VFXItem extends EffectsObject implements Disposable {
...
-   * @param classConstructor - 要添加的组件类型
+   * @param classConstructor - 要添加的组件
+   * @param options - 挂在组件上的可选参数，即会在组件初始化时传入
+   * @returns
    */
-  addComponent<T extends Component> (classConstructor: Constructor<T>): T {
-    const newComponent = new classConstructor(this.engine);
+  addComponent<T extends Component> (
+    classConstructor: Constructor<T>,
+    options: Record<string, any> = {},
```
Comment (wumaolinmaoan) [inline packages/effects-core/src/vfx-item.ts:324]: options 必须要放 在addComponent 这边吗。一般来说自定义组件是不建议通过重写构造函数初始化的，会有很多属性在构造的时候没创建。

Comment (wumaolinmaoan) [reply packages/effects-core/src/vfx-item.ts:324]: 如果是特殊的初始化逻辑的话可以自己在 Component 写一个onCreate 函数,然后链式调用addComponent(xxx).onCreate(props)。

Comment (yiiqii) [reply packages/effects-core/src/vfx-item.ts:324]: 应该也可以，我改一下

File: plugin-packages/stats/src/stats-component.ts [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+import type { StatsOptions } from './stats';
 
 export class StatsComponent extends Behaviour {
   /**
    * 监控对象
    */
   monitor: Monitor;
 
-  override onStart (): void {
+  init (options: Required<StatsOptions>): void {
```
Comment (yiiqii) [inline plugin-packages/stats/src/stats-component.ts:12]: 这块不能再用 onStart 生命周期了，看看是否有问题 @Sruimeng

**Pull request merged**